File: Leetcode/C++/561. Array Partition/solution.cpp
// https://leetcode.com/problems/array-partition/description/

#include <vector>
#include <algorithm>

using namespace std;

class Solution
{
public:
    int arrayPairSum(vector<int> &nums)
    {
        int total = 0;
        sort(nums.begin(), nums.end());
        for (int idx = 0; idx < nums.size(); idx += 2)
        {
            total += nums[idx];
        }
        return total;
    }
};

File: Leetcode/Java/561. Array Partition/solution.java
// https://leetcode.com/problems/array-partition/description/

class Solution {
    public int arrayPairSum(int[] nums) {
        Arrays.sort(nums);
        int answer = 0;

        for (int idx = 0; idx < nums.length; idx += 2){
            answer += nums[idx];
        }

        return answer;
    }
}